### DISCUSSION TITLE
Inflated Classes Full Dump

### DISCUSSION THREAD

**Discussion by SamueleD78** [vfsfitvnm/frida-il2cpp-bridge#193] [Q&A]
Again, and every time, thank you for this amazing tool.

One vital thing i do when i disassemble a game is to process the classes dump in order to create an IDA script to rename the methods calls, and it's nice and it works like a charm.

Recently, i made a script which process all the inflated classes in the classes dump and dump them in order to do the same, and this took the whole thing to the next level.

In this regard, if an inflated class isn't used as a parameter or as a variable in some other class, of course it would not be visible in the dump.

I was wondering if there is a way to extract from the game a full list of the inflated classes too, without having to guess them from the normal class dump as i'm doing now, even the ones which aren't so clearly used by other classes.

Am i dreaming? ^_^

**Comment by vfsfitvnm** [comment]
> I was wondering if there is a way to extract from the game a full list of the inflated classes too

I spent a lot of time trying to figure this out, unfortunately it doesn't seem so, e.g. you will have to guess. However, you'll realize reference types (aka classes) share the same inflated class: `Type<System.Object>`, so it's very convenient. On the other hand, value types (aka structs) have their own classes with their different method relative virtual addresses.

An idea consists in checking an inflated class existence against **every** value type and `System.Object` (or whatever reference type) for reference types.
